x-pack/plugins/security_solution/public/timelines/components/timeline/search_or_filter/index.tsx

```diff
@@ -110,19 +112,48 @@ const StatefulSearchOrFilterComponent = React.memo<Props>(
 
     const arrDataView = useMemo(() => (dataView != null ? [dataView] : []), [dataView]);
 
+    // Keep filter manager in sync with redux filters
+    useEffect(() => {
+      if (!deepEqual(filterManager.getFilters(), filters)) {
+        filterManager.setFilters(filters);
+      }
+    }, [filterManager, filters]);
+
+    // When a filter update comes in through the filter manager, update redux
+    useEffect(() => {
+      let isSubscribed = true;
+      const subscriptions = new Subscription();
+
+      subscriptions.add(
+        filterManager.getUpdates$().subscribe({
+          next: () => {
+            if (isSubscribed) {
+              const filtersWithoutDropArea = getNonDropAreaFilters(filterManager.getFilters());
+              if (!deepEqual(filtersWithoutDropArea, filters)) {
+                setFilters({
+                  id: timelineId,
+                  filters: filtersWithoutDropArea,
+                });
+              }
+            }
+          },
+        })
+      );
+
+      return () => {
+        isSubscribed = false;
+        subscriptions.unsubscribe();
+      };
+    }, [filterManager, timelineId, setFilters, filters]);
```

Comment:
It looks good, a small simplification:

```suggestion
    useEffect(() => {
      const subscription = filterManager.getUpdates$().subscribe(() => {
        if (!subscription.closed) { // <-- I don't think we need this anyway
          const filtersWithoutDropArea = getNonDropAreaFilters(filterManager.getFilters());
          if (!deepEqual(filtersWithoutDropArea, filters)) {
            setFilters({
              id: timelineId,
              filters: filtersWithoutDropArea,
            });
          }
        }
      });
      return () => {
        subscription.unsubscribe();
      };
    }, [filterManager, timelineId, setFilters, filters]);
```

Reply:
Ah, nice simplification. Yeah, the code has been extracted from a different component and is currently used as-is. I understand why we would want to check if a callback is called on an inactive subscription but semantically that shouldn't be possible.

It feels safe to assume that we don't need the `subscription.closed` check. Wdyt @semd ?

Reply:
See https://github.com/elastic/kibana/pull/180850/commits/bb32eb20f07846d7d58779c1eeac35d4f00d2f02

Reply:
Yes, I left that check for extra safety, but it feels unnecessary, the subscription execution is canceled when `unsubscribe` is called, I think we can remove it 👍

Reply:
I did a quick search for `subscription.closed` in Kibana and found only one occurrence. Other usages of `obsv.subscribe()` don't use it either.